node-repository/src/main/java/com/yahoo/vespa/hosted/provision/restapi/v2/NodesApiHandler.java

```diff
@@ -125,20 +125,21 @@ else if (path.startsWith("/nodes/v2/state/active/")) {
             return new MessageResponse("Moved " + lastElement(path) + " to active");
         }
         else if (path.startsWith("/nodes/v2/state/availablefornewallocations/")) {
-            /**
+            /*
              * This is a temporary "state" or rest call that we use to enable a smooth rollout of
              * dynamic docker flavor allocations. Once we have switch everything we remove this
-             * and change the code in the nodeadmin to delete directly (remember to allow deletion of dirty nodes then).
+             * and change the code in the nodeadmin to delete directly (remember to allow deletion of ready nodes then).
              *
              * Should only be called by node-admin for docker containers (the docker constraint is
              * enforced in the remove method)
              */
             String hostname = lastElement(path);
+            nodeRepository.setReady(hostname);
```

Comment:
Instead, pass an argument to removeRecursively to allow removal of dirty docker container.

Reply:
This will clutter the API with the option to delete from a state that you're not supposed to be able to?

Reply:
Injecting a move to ready just before removal seems much worse IMO.